### DISCUSSION TITLE
Rack Import Server Error

### DISCUSSION THREAD

**Discussion by yhiciano** [netbox-community/netbox#11876] [Q&A] [answered]
### NetBox version

v3.4.5

### Python version

3.8

### Steps to Reproduce

1. Rack Bulk Import
2. Upload CSV
3. Submit

### Expected Behavior

Import a list of racks.

### Observed Behavior

The list of racks was not imported, instead got this error. If I don't use the site column, it gets through but for my personal implementation the site is a key value
![image](https://user-images.githubusercontent.com/125702895/222247275-56fadc73-d0e8-4840-981a-93591efb7c94.png)

**Comment by candlerb** [comment]
"If I don't use the site column, it gets through but for my personal implementation the site is a key value"

Sorry, I don't understand that.

Every Rack must belong to a Site - as the error says.  Please can you give a sample of the CSV you are trying to upload, with both the header row and one data row?

**Comment by yhiciano** [comment]
What i mean is that apparently is not a requirement, and if I delete that column, then all the racks that share the same name because they belong to different sites, when i import them it says they are duplicate (but with the site column there wouldn't be duplicate values)
I've attached the same exact csv i'm trying to import.
[netbox_racks.csv](https://github.com/netbox-community/netbox/files/10871408/netbox_racks.csv)

**Comment by candlerb** [comment]
Sorry, I still don't see the problem.

If I import the following:

```
site,name,status,width,u_height
,testing,active,19,42
```

then Netbox gives me a Server Error and exception ("Rack has no site").  [Aside: it ought to report it properly, not with an exception]

If I remove the site column completely, i.e.

```
name,status,width,u_height
testing,active,19,42
```

then I get the same error.

If have an existing rack called "test" in a site called "Test", then I try to import

```
name,status,width,u_height
test,active,19,42
```

then I still get the same error ("Rack has no site").  However if I try to import:

```
site,name,status,width,u_height
Test,test,active,19,42
```

then I get a duplicate rack, i.e. another rack with name "test". Maybe that's what you're talking about?  But that's not an issue with CSV upload: using the GUI, I can also create multiple racks all called "test" in the same site "Test".  (Looking at the [model code](https://github.com/netbox-community/netbox/blob/v3.4.5/netbox/dcim/models/racks.py#L49), there's no unique constraint on either name or facility_id - but there is on asset_tag)

However, I cannot see how to import a rack without a site via CSV upload, which is what you seem to be saying you can do.

**Reply by yhiciano** [reply]
> then Netbox gives me a Server Error and exception ("Rack has no site"). [Aside: it ought to report it properly, not with an exception]

Exactly but the rack does have the site values in the csv. I don't know why i get that error.

> then I get a duplicate rack, i.e. another rack with name "test". Maybe that's what you're talking about?

Yes that's what i mean. When i removed the site column it didn't ask for the site column and just printed the duplicate rack message. I just tried doing it to see if i could get rid of the rack has no site message.



My ultimate goal is to upload the csv that have attached, so i don't have to do it manually.
[netbox_racks.csv](https://github.com/netbox-community/netbox/files/10872878/netbox_racks.csv)

**Reply by candlerb** [reply]
> When i removed the site column it didn't ask for the site column and just printed the duplicate rack message.

Sorry, I still don't get it.  If you can give a sequence of steps to reproduce the problem from scratch, then maybe someone can look at it.

**Reply by yhiciano** [reply]
> Sorry, I still don't get it. If you can give a sequence of steps to reproduce the problem from scratch, then maybe someone can look at it.

Sorry I'm not able to recreate it anymore. But really the problem is that i get the no site error when in fact it has a site value.

**Comment by yhiciano** [comment, marked as answer]
I found the issue. One of the sites was not created. I could import without issues. Thanks
